### DISCUSSION TITLE
Communication between Godot editor and the game running

### DISCUSSION THREAD

**Discussion by pixel-boy** [godotengine/godot-proposals#10994] [Editor]
In my game I have debug tools that are available via an in-game interface, it allows me among other things to quickly launch a scene by selecting from a list.

I would like to be able to open scene / resource in the editor via a button that would be in the in-game interface.

The **EditorInterface** class allows you to open a file according to a **file_path** but the class is only available via plugins and not when the game is running.

I'm not sure if there is a way to do this technically, but It could be handy so I'm throwing the idea out there! I'm also open to any workaround suggestions.

**Comment by dalexeev** [comment]
This is already possible. See:

* [`EngineDebugger`](https://docs.godotengine.org/en/stable/classes/class_enginedebugger.html)
* [`EditorDebuggerPlugin`](https://docs.godotengine.org/en/stable/classes/class_editordebuggerplugin.html)
* [`EditorDebuggerSession`](https://docs.godotengine.org/en/stable/classes/class_editordebuggersession.html)

**Reply by pixel-boy** [reply]
Ah perfect, I'll take a look at that, thanks!

**Reply by pixel-boy** [reply]
I tried to do what I described above, but I can't seem to send a message from the game to the **EditorDebuggerPlugin**.
The **EngineDebugger** class is supposed to be a link to the **EditorDebuggerPlugin** ? 

When I do:
`if EngineDebugger.has_capture("open_scene"):`
I don't get any **"_has_capture()"** calls from my **EditorDebuggerPlugin** class side.

The documentation is pretty light on the subject, I wonder if I'm missing something.

**Reply by dalexeev** [reply]
> I tried to do what I described above, but I can't seem to send a message from the game to the **EditorDebuggerPlugin**.

From the game you should send a message using `EngineDebugger` and listen it in `EditorDebuggerPlugin`.

**Reply by pixel-boy** [reply]
In my in game debug interface I do:
`EngineDebugger.send_message("open_scene",[scene_path])`
But I get the error "Invalid message received".

And it's return false if I write:
`if EngineDebugger.has_capture("open_scene"):`

In the **EditorDebuggerPlugin** I have:
```
func _has_capture(prefix):
	print("test")
	return prefix == "open_scene"
```
But the function is never called.

**Edit:** 
Add the line:
` EngineDebugger.register_message_capture("open_scene",on_open_scene)`
Seems to make **has_capture()** returning **true**, but I still get the error "Invalid message received" when sending the message.

**Edit 2:**
It seems the message should end with ":" ( "open_scene:" in my case) to be valid, the error may have come from there.

**Reply by dalexeev** [reply]
> `EngineDebugger.send_message("open_scene",[scene_path])`

The message should consist of two parts separated by a colon (a prefix and some identifier), like `my_plugin:some_action`. Also note that while the game is running, `print()` in the editor does not output anything.

I am absolutely sure that this works, as I tested it yesterday. But I am not at my desktop right now, so I can not provide you with a working example. You can search for examples on GitHub or ask for help in the community channels.

**Reply by pixel-boy** [reply]
Thanks for your help! The problem was indeed with the syntax.

The documentation doesn't talk about this aspect very clearly, even if the examples do follow this format, maybe it could be relevant to add a comment about this on the documentation page ? Or is there a better place to report this?

**Reply by dalexeev** [reply]
Yes, I also noticed a couple of errors in the documentation (missing `return true`, no mention of what you have to do on the game side). If you want to contribute, you can open a PR in the main repository and link directly to this discussion. The documentation bug reports are scattered across several repositories (godot, godot-docs, godot-proposals). We improve the documentation when we see users encountering problems, it is not really matter on what platform it was initially reported.

**Reply by dalexeev** [reply]
* See https://github.com/godotengine/godot/pull/98378.
